### PR TITLE
Oliinyk / Region and Area admins integration

### PR DESCRIPTION
**Pull request by doliinyk**

### CODE REVIEW

**Comment by doliinyk** [comment]
Also waiting from backend:
- fix fetching area admins by region admin (now return 400)
- filtering admins due to user institution/CATOTTG on backend

File: src/app/shared/store/admin.actions.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export class GetProviderAdminHistory {
 export class BlockProviderById {
   static readonly type = '[admin] Block Provider';
 
-  constructor(public payload: BlockProviderData, public parameters: MinistryAdminParameters) {}
+  constructor(public payload: BlockProviderData, public parameters: ProviderParameters) {}
```
Comment (doliinyk) [inline src/app/shared/store/admin.actions.ts:174]: Not sure about this change

File: src/app/shared/store/registration.state.ts [hunk 2]
```diff
@@ -179,41 +182,40 @@ export class RegistrationState {
...
+        return this.regionAdminService.getAdminProfile().pipe(tap((regionAdmin: RegionAdmin) => patchState({ regionAdmin: regionAdmin })));
       case Role.ministryAdmin:
         return this.ministryAdminService
           .getAdminProfile()
           .pipe(tap((ministryAdmin: MinistryAdmin) => patchState({ ministryAdmin: ministryAdmin })));
       case Role.areaAdmin:
         return this.territorialCommunityAdmin
           .getAdminProfile()
-          .pipe(tap((territorialCommunityAdmin: TerritorialCommunityAdmin) => patchState({ territorialCommunityAdmin: territorialCommunityAdmin })));
+          .pipe(tap((territorialCommunityAdmin: TerritorialCommunityAdmin) => patchState({ areaAdmin: territorialCommunityAdmin })));
```
Comment (humanmantis) [inline src/app/shared/store/registration.state.ts]: Rename `territorialCommunityAdmin` to `areaAdmin` for consistency

File: src/app/shell/admin-tools/admin-tools.guard.ts [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
   canLoad(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
-    return this.role$.pipe(
-      filter((role: string) => role === Role.techAdmin || role === Role.ministryAdmin),
-      map((role: string) => role === Role.techAdmin || role === Role.ministryAdmin)
-    );
+    return this.role$.pipe(filter(Boolean), map(isRoleAdmin));
   }
 }
+
+export function isRoleAdmin(role: string): boolean {
```
Comment (humanmantis) [inline src/app/shell/admin-tools/admin-tools.guard.ts]: Move these functions to utils

File: src/app/shared/store/registration.state.ts [hunk 4]
```diff
@@ -1,57 +1,56 @@
...
+  isAuthorizationLoading: boolean;
   user: User;
   provider: Provider;
   parent: Parent;
   techAdmin: TechAdmin;
-  regionAdmin: RegionAdmin;
   ministryAdmin: MinistryAdmin;
-  territorialCommunityAdmin: TerritorialCommunityAdmin;
+  regionAdmin: RegionAdmin;
+  areaAdmin: TerritorialCommunityAdmin;
```
Comment (humanmantis) [inline src/app/shared/store/registration.state.ts]: You add `areaAdmin` here, but do not replace `territorialCommunityAdmin` below (line 216)

**Review by humanmantis — CHANGES_REQUESTED** [review]

**Review by humanmantis — APPROVED** [review]

**Pull request merged**